### DISCUSSION TITLE
How to get all the nodes of an editor's state with customPlugin ?

### DISCUSSION THREAD

**Discussion by Michael-indochat** [facebook/lexical#4903] [Q&A]
I want to create a plugin that can retrieve the updated information of all nodes each time I make changes in the editor. My goal is to perform a character count of these nodes, while excluding special nodes called "mentionNode."

For instance, if there's a sentence in the editor that says "Hello, @FriendName," where "@FriendName" is a mentionNode,
I want to calculate the character count while disregarding this node.
Instead, I would only count the characters in the "Hello," part of the text.

**Comment by Michael-indochat** [comment]
At present, my judgment is that I can get AllNode, and judge whether each node is a mentionNode `$isMentionNode` through the recursive syntax, and calculate the amount of content with the result

**Comment by Alex-ray** [comment]
You can use [`$nodesOfType`](https://lexical.dev/docs/api/modules/lexical#nodesoftype)!

```
import { $nodesOfType } from "lexical";

editorState.read(() => {
    const listNodes = $nodesOfType(ListNode);
})
```

**Comment by CanRau** [comment]
There's also `$dfs`, `$getChildrenRecursively` and more at https://github.com/facebook/lexical/discussions/4774#discussioncomment-7998448
